python-backend/app/api/mines/variances/models/variance.py

```diff
@@ -39,7 +39,8 @@ class Variance(AuditMixin, Base):
     received_date = db.Column(db.DateTime, nullable=False)
     expiry_date = db.Column(db.DateTime)
 
-    documents = db.relationship('MineDocument', lazy='joined', secondary='variance_document_xref')
+    documents = db.relationship('VarianceDocumentXref', lazy='joined')
```

Comment:
I don't think this attribute is necessary .

Reply:
This attribute is necessary because I want to marshal the `VarianceDocumentXref` model now, not the `MineDocument` model because `MineDocument` doesn't have the `uploaded_at` attribute.

As I mentioned in the opening PR comment, the business case for `uploaded_at` is when the mine_document becomes associated with the variance. So, it needs to (as far as I can tell) live on `variance_document_xref`.

Does that make sense?

Reply:
if that's the case then the column title is a bit unclear. how about `attached_at` or `created_at`

Reply:
`created_at` is probably a better name because it reflects both the implementation details and the business use case.

The reason for `uploaded_at` is/was that the business wants "the time the document was uploaded to the variance." I think that `uploaded_at` reflects that use case. It's not the businesses fault that we allow document uploads to a non-existent variance. Which is a process/user flow that I think should be changed and is currently being investigated by the UX team.

Point being, if you don't feel good about `uploaded_at`, I can rename it to `created_at`, but I also think the concern with `uploaded_at` is a symptom of weird implementation details that don't necessarily reflect the end-user's goal(s) / use case.

Reply:
documents may live in the document manager before they are added to the api, but the record in mine_documents shouldn't exist until the variance does. I haven't looked at the variance implementation, i'm referring to the permit/permit amendment document workflow. the odd thing is that `VarianceDocumentXref`s aren't 'uploaded', so `created_at` seems like a better name.